agent/src/util/slot_reconciliation.rs

```diff
@@ -163,16 +170,31 @@ impl DevicePluginSlotReconciler {
                 .device_usage
                 .iter()
                 .filter_map(|(k, v)| {
-                    if v != node_name && node_slot_usage.contains(k) {
-                        // We need to add node_name to this slot IF
-                        //     the slot is not labeled with node_name AND
-                        //     there is a container using that slot on this node
-                        Some(k.to_string())
-                    } else {
+                    let same_node_name = match DeviceUsage::from_str(v) {
```

Comment:
would this be more dogmatic rust if it were:

```rust
                    match DeviceUsage::from_str(v) {
                        Ok(usage) => if usage.is_same_usage(node_name) {
                            None
                        } else {
                            // We need to add node_name to this slot IF
                            //     the slot is not labeled with node_name AND
                            //     there is a container using that slot on this node
                            node_slot_usage
                                .iter()
                                .find(|u| match DeviceUsage::from_str(u) {
                                    Ok(slot_usage) => slot_usage.is_same_usage(k),
                                    Err(_) => false,
                                })
                                .map(|usage| {
                                    (
                                        k.to_string(),
                                        DeviceUsage::from_str(usage).unwrap().get_kind(),
                                    )
                                })
                        },
                        Err(_) => None,
                    };
```

Reply:
we want to treat it as different usage if the first parsing DeviceUsage::from_str(v) returns error.  The proposed code treats it as the same usage.
anyway, I'll refactor the code.

Reply:
yup, my code is definitely wrong ... sorry!

i think my code suggestion should've been something closer to this:

```rust
if match DeviceUsage::from_str(v) {
    Ok(usage) => usage.is_same_usage(node_name),
    Err(_) => false,
} {
    None
} else {
    // We need to add node_name to this slot IF
    //     the slot is not labeled with node_name AND
    //     there is a container using that slot on this node
    node_slot_usage
        .iter()
        .find(|u| match DeviceUsage::from_str(u) {
            Ok(slot_usage) => slot_usage.is_same_usage(k),
            Err(_) => false,
        })
        .map(|usage| {
            (
                k.to_string(),
                DeviceUsage::from_str(usage).unwrap().get_kind(),
            )
        })
}
```